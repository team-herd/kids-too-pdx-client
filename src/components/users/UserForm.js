import React from 'react';
import PropTypes from 'prop-types';

function UserForm({ handleSubmit, handleChange }) {
  return (
    <form onSubmit={handleSubmit}>
      <fieldset>
        <legend>User Info</legend>
        <label>Email:
          <input type="text" name="email" onChange={handleChange} required />
        </label>

        <label>Password:
          <input type="text" name="password" onChange={handleChange} required />
        </label>
      </fieldset>
    </form>
  );
}

UserForm.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  user: PropTypes.object
};

export default UserForm;
