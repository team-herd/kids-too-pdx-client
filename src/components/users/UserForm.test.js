import React from 'react';
import renderer from 'react-test-renderer';
import UserForm from './UserForm';

describe('Sign up form', () => {
  it('snapshots UserForm', () => {
    const props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      user: {
        email: 'theOrg@org.com',
        password: 'pass'
      }
    };

    const tree = renderer.create(
      <UserForm {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
