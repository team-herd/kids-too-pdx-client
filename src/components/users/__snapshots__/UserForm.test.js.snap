// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up form snapshots UserForm 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      User Info
    </legend>
    <label>
      Email:
      <input
        name="email"
        onChange={[MockFunction]}
        required={true}
        type="text"
        value="theOrg@org.com"
      />
    </label>
    <label>
      Password:
      <input
        name="password"
        onChange={[MockFunction]}
        required={true}
        type="text"
        value="pass"
      />
    </label>
  </fieldset>
</form>
`;
